### PR TITLE
feat: Add url path redirect rules

### PR DESCRIPTION
**Pull request by katie-gardner**
### Overview
The current redirect rules are only for host redirects, and don't support rerouting based on specific url path. To avoid an impact on any existing usage of the rule set, this adds another rule set for url redirection with all possible parameters added as configurable options. 

### Other finding
This also fixes a small bug with the cdn frontdoor route: with redirects specified the tf failed to apply with an error in the rule set ids `The 'true' value is string, but the 'false' value is tuple.` The redirect id just needed to be in a list.

### CODE REVIEW

File: cdn.tf [hunk 1]
```diff
@@ -131,6 +139,48 @@ resource "azurerm_cdn_frontdoor_rule" "redirect" {
...
+      query_string         = lookup(host_redirect, "query_string", null),
+      operator             = host_redirect.operator,
+      match_values         = lookup(host_redirect, "match_values", null),
+      transforms           = lookup(host_redirect, "transforms", null)
+    }
+  } : {}
+
+  depends_on = [azurerm_cdn_frontdoor_origin_group.waf, azurerm_cdn_frontdoor_origin.waf]
+
+  name                      = "urlredirect${each.key}"
```
Comment (jimwashbrook) [inline cdn.tf:159]: Is it worth changing this to be something more descriptive, e.g. based off one of the redirect values?

File: variables.tf [hunk 2]
```diff
@@ -182,6 +182,22 @@ variable "cdn_host_redirects" {
   default     = []
 }
 
+variable "cdn_url_path_redirects" {
```
Comment (jimwashbrook) [inline variables.tf:185]: Is it worth adding any validation to this or just leaving it for the user to implement correctly?

**Review by jimwashbrook — APPROVED** [review]

File: variables.tf (hunk 2 above)
Comment (katie-gardner) [reply variables.tf:185]: As talked about since the validation comes with `azurerm` out of the box, I'll leave this one

File: cdn.tf (hunk 1 above)
Comment (katie-gardner) [reply cdn.tf:159]: The only non-nullable ones are operator & request type, I could add these but I'm not sure they'll add much over the standard `rule{x}`

**Review by jack-coggin — APPROVED** [review]

File: variables.tf [hunk 3]
```diff
@@ -182,6 +182,22 @@ variable "cdn_host_redirects" {
...
+  description = "CDN FrontDoor url path redirects `[{ \"redirect_type\": \"PermanentRedirect\", \"destination_path\": \"/example\", \"destination_hostname\": \"www.example.uk\", \"operator\": \"Equals\", \"match_values\": [\"/example\"] }]`"
+  type = list(object({
+    redirect_type        = string
+    redirect_protocol    = optional(string)
```
Comment (DrizzlyOwl) [inline variables.tf:189]: If you change these to `optional(string, null)` or `optional(list(string), [])` then you don't need to `lookup()` the values later on

**Review by DrizzlyOwl — CHANGES_REQUESTED** [review]
I would like to see the `lookup`s squashed by setting the defaults properly, if thats possible. Unless I'm missing something obvious, in which case please let me know

File: variables.tf (hunk 3 above)
Comment (katie-gardner) [reply variables.tf]: Fixed - thanks for highlighting this

**Review by DrizzlyOwl — APPROVED** [review]

**Pull request merged**